pkg/gardenlet/controller/backupbucket/reconciler.go

```diff
@@ -350,45 +350,52 @@ func (r *Reconciler) reconcileBackupBucketExtensionSecret(ctx context.Context, e
-	var gardenGeneratedSecretRef *corev1.SecretReference
-
-	if extensionBackupBucket.Status.GeneratedSecretRef != nil {
-		seedGeneratedSecret, err := kubernetesutils.GetSecretByReference(ctx, r.SeedClient, extensionBackupBucket.Status.GeneratedSecretRef)
-		if err != nil {
-			return err
-		}
-
-		gardenGeneratedSecret := &corev1.Secret{
+	var (
+		gardenGeneratedSecret = &corev1.Secret{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      generateGeneratedBackupBucketSecretName(backupBucket.Name),
 				Namespace: r.GardenNamespace,
 			},
 		}
-		ownerRef := metav1.NewControllerRef(backupBucket, gardencorev1beta1.SchemeGroupVersion.WithKind("BackupBucket"))
-
-		if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, r.GardenClient, gardenGeneratedSecret, func() error {
-			gardenGeneratedSecret.OwnerReferences = []metav1.OwnerReference{*ownerRef}
-			controllerutil.AddFinalizer(gardenGeneratedSecret, finalizerName)
-			gardenGeneratedSecret.Data = seedGeneratedSecret.DeepCopy().Data
-			return nil
-		}); err != nil {
-			return err
-		}
+		gardenGeneratedSecretRef *corev1.SecretReference
+	)
 
+	if extensionBackupBucket.Status.GeneratedSecretRef != nil {
 		gardenGeneratedSecretRef = &corev1.SecretReference{
 			Name:      gardenGeneratedSecret.Name,
 			Namespace: gardenGeneratedSecret.Namespace,
 		}
 	}
 
-	if gardenGeneratedSecretRef != nil || extensionBackupBucket.Status.ProviderStatus != nil {
+	if extensionBackupBucket.Status.GeneratedSecretRef != nil || extensionBackupBucket.Status.ProviderStatus != nil {
 		patch := client.MergeFrom(backupBucket.DeepCopy())
+		// Update the BackupBucket status here before going for the CreateOrGetAndStrategicMergePatch call, so that the SeedAuthorizer
+		// can add the entry for this secret in the graph. See https://github.com/gardener/gardener/issues/7705 for more details.
 		backupBucket.Status.GeneratedSecretRef = gardenGeneratedSecretRef
 		backupBucket.Status.ProviderStatus = extensionBackupBucket.Status.ProviderStatus
 		if err := r.GardenClient.Status().Patch(ctx, backupBucket, patch); err != nil {
 			return err
 		}
 	}
+
+	if extensionBackupBucket.Status.GeneratedSecretRef != nil {
+		seedGeneratedSecret, err := kubernetesutils.GetSecretByReference(ctx, r.SeedClient, extensionBackupBucket.Status.GeneratedSecretRef)
+		if err != nil {
+			return err
+		}
+
+		if _, err := controllerutils.CreateOrGetAndStrategicMergePatch(ctx, r.GardenClient, gardenGeneratedSecret, func() error {
+			gardenGeneratedSecret.OwnerReferences = []metav1.OwnerReference{
+				*metav1.NewControllerRef(backupBucket, gardencorev1beta1.SchemeGroupVersion.WithKind("BackupBucket")),
+			}
+			controllerutil.AddFinalizer(gardenGeneratedSecret, finalizerName)
+			gardenGeneratedSecret.Data = seedGeneratedSecret.DeepCopy().Data
+			return nil
+		}); err != nil {
+			return err
+		}
+	}
```

Comment:
```suggestion
	if extensionBackupBucket.Status.GeneratedSecretRef != nil {
		gardenGeneratedSecretRef = &corev1.SecretReference{
			Name:      gardenGeneratedSecret.Name,
			Namespace: gardenGeneratedSecret.Namespace,
		}

		patch := client.MergeFrom(backupBucket.DeepCopy())
		// Update the BackupBucket status here before going for the CreateOrGetAndStrategicMergePatch call, so that the SeedAuthorizer
		// can add the entry for this secret in the graph. See https://github.com/gardener/gardener/issues/7705 for more details.
		backupBucket.Status.GeneratedSecretRef = gardenGeneratedSecretRef
		if err := r.GardenClient.Status().Patch(ctx, backupBucket, patch); err != nil {
			return err
		}

		seedGeneratedSecret, err := kubernetesutils.GetSecretByReference(ctx, r.SeedClient, extensionBackupBucket.Status.GeneratedSecretRef)
		if err != nil {
			return err
		}

		if _, err := controllerutils.CreateOrGetAndStrategicMergePatch(ctx, r.GardenClient, gardenGeneratedSecret, func() error {
			gardenGeneratedSecret.OwnerReferences = []metav1.OwnerReference{
				*metav1.NewControllerRef(backupBucket, gardencorev1beta1.SchemeGroupVersion.WithKind("BackupBucket")),
			}
			controllerutil.AddFinalizer(gardenGeneratedSecret, finalizerName)
			gardenGeneratedSecret.Data = seedGeneratedSecret.DeepCopy().Data
			return nil
		}); err != nil {
			return err
		}
	}
	
	if extensionBackupBucket.Status.ProviderStatus != nil {
		patch := client.MergeFrom(backupBucket.DeepCopy())
		backupBucket.Status.ProviderStatus = extensionBackupBucket.Status.ProviderStatus
		if err := r.GardenClient.Status().Patch(ctx, backupBucket, patch); err != nil {
			return err
		}
	}
```

This way, we'll have two status patches, but I think it's easier to read/more clear what happens.